pynamodb/models.py

```diff
@@ -675,6 +675,26 @@ def scan(cls,
             rate_limit=rate_limit,
         )
 
+    @classmethod
+    def record_exists(cls, hash_key, range_key=None):
+        """
+        Checks if the given hash_key and optional range_key pair exists in the table.
+
+        This will return true if at least one record matching hash_key and range_key exists
+
+        :param hash_key: The hash key to look up
+        :param range_key: If set, include in lookup
+        """
+        try:
+            if range_key is None:
+                cls.query(hash_key).next()
+            else:
+                cls.query(hash_key, cls._range_key_attribute() == range_key).next()
```

Comment:
```suggestion
            if range_key is None:
                next(cls.query(hash_key, page_size=1))
            else:
                next(cls.query(hash_key, cls._range_key_attribute() == range_key, page_size=1))
```

1. I recall restricting `page_size` leads to spending fewer RCUs. @garrettheel?
2. Total nit, but maybe using `next` built-in is more idiomatic?

Reply:
Sounds good to me, I applied your suggestion.